lib/cocoapods/sources_manager.rb

```diff
@@ -192,6 +214,44 @@ def search_index_path
       extend Executable
       executable :git
 
+      # Updates the stored search index if there are changes in spec repos while updating them.
+      # Update is performed incrementally. Only the changed pods' search data is re-generated and updated.
+      # @param  [Hash{Source => Array<String>}] changed_spec_paths
+      #                  A hash containing changed specification paths for each source.
+      def update_search_index_if_needed(changed_spec_paths)
+        search_index = nil
+        changed_spec_paths.each_pair do |source, spec_paths|
+          source_name = source.name
+          next unless spec_paths.length > 0
+          search_index = stored_search_index
```

Comment:
We do not want to create a search index from scratch while updating repos if there is no search index created before. So i think it should stay as it is, and exit from block at line 229.